light-client/src/predicates.rs

```diff
@@ -73,12 +73,10 @@ pub trait VerificationPredicates: Send {
         &self,
         signed_header: &SignedHeader,
         validators: &ValidatorSet,
-        validator: &dyn CommitValidator,
+        commit_validator: &dyn CommitValidator,
     ) -> Result<(), VerificationError> {
-        // FIXME: Do not discard underlying error
-        validator
-            .validate(signed_header, validators)
-            .map_err(|e| VerificationError::InvalidCommit(e.to_string()))?;
+        commit_validator.validate(signed_header, validators)?;
+        commit_validator.validate_full(signed_header, validators)?;
```

Comment:
Is there actually a place where `commit_validator.validate` is used without `validate_full`?

Reply:
No there's not. We currently perform both checks, and I don't see the need/place for doing just `validate` since we do 2/3 full verification of every block by that point. But perhaps that's not right, and we need to only to `validate_full` somewhere here: https://github.com/informalsystems/tendermint-rs/pull/358/files/b1aee8d1f4b42d56a5e43dda9fc52572455ff4ea#diff-1911c69336354ecbef5a4524721976faR236-R263

Reply:
if we do https://github.com/informalsystems/tendermint-rs/issues/377, we can prob get rid of validate_full, especially since we're trying not to rely on the underlying validator_address. though I suppose so long as it is there it should be checked.